src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -125,14 +126,26 @@ private static int getLvIndex(String desc, boolean isStatic, int asmIndex) {
 		}
 
 		@Override
-		public void visitParameter(String name, int access) {
-			name = ((AsmRemapper) remapper).mapMethodArg(this.owner, this.name, this.desc, getLvIndex(this.desc, isStatic, argsVisited), name);
-			argsVisited++;
-			super.visitParameter(name, access);
+		public void visitParameter(final String name, int access) {
+			String newName = ((AsmRemapper) remapper).mapMethodArg(this.owner, this.name, this.desc, getLvIndex(this.desc, isStatic, argsVisited), name);
+
+			if (newName == null || newName.equals(name)) {
+				newName = renamedInvalidLocals.get(argsVisited);
+			}
+
+			if(this.isAbstract && (newName == null || newName.equals(name))) {
+				newName = "p_" + nameCounts.compute("p", (k, v) -> (v == null) ? 1 : v + 1);
+			}
+
+			if(newName != null) {
```

Comment:
For names carried over directly from the class file which aren't immediately remapped or renamed by the LVT (presuming a non-abstract method), it would be nice to keep the access modifier they might have even if the name is thrown away

Reply:
In the newest commit, it keeps it always, when renameinavlidmodifier is not set. Otherwise, it only keeps it when we found a valid identifier.
I am not sure how to keep the access modifier otherwise, perhabs just push it to a list every time the method is called and later retrieve it from there?

Reply:
Until `checkedParams` is `true` you can be sure that the visits are coming from the `ClassReader` rather than internally, at which point if `access != 0` it could be stored in an `int[]` with length `argCount`.

Then once `checkedParams` is `true`, `#checkParameters` could pull the access from the array rather than always using `0`. Although this might need a separate way of watching the end of the array up to this point if `argsVisited` isn't always bumped.

Reply:
I'm now using a different approach where the parameter is guaranteed when the option is set, so this is no longer a problem.